klab.authentication/src/main/java/org/integratedmodelling/klab/Authentication.java

```diff
@@ -418,8 +421,35 @@ public IUserIdentity authenticate(ICertificate certificate) throws KlabAuthoriza
         return ret;
     }
 
-    public ExternalAuthenticationCredentials getCredentials(String hostUrl) {
-        return externalCredentials.get(hostUrl);
+    /**
+     * Returns a credential for the selected host. It can be an ID or an URL as long as it is presented in the credentials.
+     * @param endpoint as a URL or an ID defined in the credentials
+     * @return
+     */
+    public ExternalAuthenticationCredentials getCredentials(String endpoint) {
+        if (endpoint.startsWith("http://") || endpoint.startsWith("https://")) {

```

Comment:
Can we have an authentication endpoint over `http`? 🤔

Reply:
This definition is supposed to be generic so I would argue that communication over `http://` should be supported.
An example of mine: Not long ago I was testing an application via `http://localhost:9000/my-app` and I needed to use credentials.